services/mfa/mfa.go

```diff
@@ -111,17 +111,16 @@ func (m *Mfa) Deactivate(userId string) *model.AppError {
 		return err
 	}
 
+	if err := m.Store.User().UpdateMfaSecret(userId, ""); err != nil {
+		return model.NewAppError("Deactivate", "mfa.deactivate.save_secret.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
 	achan := m.Store.User().UpdateMfaActive(userId, false)
-	schan := m.Store.User().UpdateMfaSecret(userId, "")
```

Comment:
This was async on purpose, the idea here would be to use the explicit async approach creating the channel, and running it inside a gorouting (you can see an example in the example PR).

Reply:
Thank you for the review @jespino! I have changed things around to mirror the example PR